CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.0)
+
+include("cmake/HunterGate.cmake")
+HunterGate(
+  URL "https://github.com/ruslo/hunter/archive/v0.12.44.tar.gz"
+  SHA1 "54963db4cd6e2a8780643a3cc48bdbbefedec6cc"
+)
+
+hunter_add_package(ZLIB)
```

Comment:
Funny... I tested the package on travis and appveyor, before any modifications. BUT you are right I cannot build Protobuf as a program that use hunter.
It works if you use it inside hunter (so it is built as a packaged installed by hunter) and if you build it as a program with the modified cmake/CMakeLists.txt directly.
You can see the test results:
https://travis-ci.org/tatraian/hunter_test_package/builds/114457180
https://ci.appveyor.com/project/tatraian/hunter-test-package/build/1.0.11
(yes, Windows doesn't like me...).

An extra surprise that clang 3.4 compiles protobuf but clang 3.6 doesn't. If my fix is rejected by Google I would like to put it in this repository. (https://github.com/google/protobuf/pull/1299)

But go back to the original problem. Unfortunately I couldn't make Protobuf be hunter based. Since add_subdirectory disturbs project detection and version setup. So the next scenario doesn't work:
```
#CMakeLists.txt
cmake_minimum_required(VERSION 3.0)

include("cmake/HunterGate.cmake")
HunterGate(
   ...
)

add_subdirectory(cmake)

#cmake/CMakeLists.txt
project(protobuf C CXX)

hunter_add_package(ZLIB)

...
```

I always get "huntergate must be before project" error. And I got the same if I move everything in cmake/CMakeLists.txt except add_subdirectory. So I can't do this better, sorry.